core-bundle/src/Routing/ResponseContext/ResponseContextAccessor.php

```diff
@@ -0,0 +1,53 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of Contao.
+ *
+ * (c) Leo Feyer
+ *
+ * @license LGPL-3.0-or-later
+ */
+
+namespace Contao\CoreBundle\Routing\ResponseContext;
+
+use Symfony\Component\HttpFoundation\RequestStack;
+
+class ResponseContextAccessor
```

Comment:
I find this class name rather strange. Also, shouldn't these methods just live the factory? So you only have one class to get/set/create a response context?

Reply:
I thought about that too but I decided against it because as a user you are normally only accessing the current context. It would be super strange to inject a factory to access the context. That's why they are split and I like that better.